src/odb/src/zutil/util.cpp

```diff
@@ -49,15 +49,19 @@ using std::vector;
 RUDYCalculator::RUDYCalculator(dbBlock* block) : block_(block)
 {
   gridBlock_ = block_->getDieArea();
+  if (gridBlock_.area() == 0) {
+    return;
+  }
   // TODO: Match the wire width with the paper definition
   wireWidth_ = block_->getTech()->findRoutingLayer(1)->getWidth();
 
   odb::dbTechLayer* tech_layer = block_->getTech()->findRoutingLayer(3);
   odb::dbTrackGrid* track_grid = block_->findTrackGrid(tech_layer);
+  if (track_grid == nullptr) {
```

Comment:
Similar comment here. I believe that if a layer doesn't have a track grid pattern defined at the stage RUDY is called, we should throw an error.